pkg/trace/agent/agent.go

```diff
@@ -584,37 +615,32 @@ func (a *Agent) getAnalyzedEvents(pt *traceutil.ProcessedTrace, ts *info.TagStat
 	return events
 }
 
-// runSamplers runs all the agent's samplers on pt and returns the sampling decision
-// along with the sampling rate.
+// runSamplers runs the agent's priority sampler on pt and returns the sampling decision along with
+// the sampling rate.
+//
+// The rare sampler is run first, catching all rare traces early. If the trace has a priority set,
+// the sampling priority is used with the Priority Sampler. When there is no priority set, the
+// NoPrioritySampler is run. Finally, if the trace has not been sampled by the other samplers, the
+// error sampler is run.
```

Comment:
The fact that this function and `runProbabilisticSampler` duplicate some code (calling rare and error samplers) is hurting the readability and the maintainability. A possible alternative is to consolidate all sampler calls in `runSamplers`.

I suggest renaming the function `runSamplers` to `runPrioritySampler` for clarity if we decide to keep the logic as is.

Reply:
I merged the functions into one, but I'm not sure it's simpler, given the logic. Let me know what you think.